# WGA Analysis 


This notebook is the data analysis of the website [Web Gallery of Art](https://www.wga.hu/). 


## Imports

Important library imports 

In [3]:
import pandas as pd

## Data Analysis

Using the [dataset](https://www.wga.hu/database/download/index.html) from the Web Gallery of Art, it was put in the dataframe below to show its contents. 

In [4]:
table = pd.read_excel("catalog.xlsx")
table

AUTHOR  \
0              AACHEN, Hans von   
1              AACHEN, Hans von   
2              AACHEN, Hans von   
3              AACHEN, Hans von   
4              AACHEN, Hans von   
...                         ...   
51454             ZÜRN, Michael   
51455  ZWIRNER, Ernst Friedrich   
51456  ZWIRNER, Ernst Friedrich   
51457  ZWIRNER, Ernst Friedrich   
51458  ZWIRNER, Ernst Friedrich   

                                               BORN-DIED  \
0                        (b. 1552, Köln, d. 1615, Praha)   
1                        (b. 1552, Köln, d. 1615, Praha)   
2                        (b. 1552, Köln, d. 1615, Praha)   
3                        (b. 1552, Köln, d. 1615, Praha)   
4                        (b. 1552, Köln, d. 1615, Praha)   
...                                                  ...   
51454  (b. ca. 1626, Wasserburg am Inn, d. ca. 1691, ...   
51455              (b. 1802, Jakobswalde, d. 1861, Köln)   
51456              (b. 1802, Jakobswalde, d. 1861, Köln)   
51457              (b. 1802, Jakobswalde, d. 1861, Köln)   
51458              (b. 1802, Jakobswalde, d. 1861, Köln)   

                                      TITLE     DATE  \
0                          Venus and Adonis  1574-88   
1                                  Allegory     1598   
2      Allegory of Peace, Art and Abundance     1602   
3                Jupiter, Antiope and Cupid  1595-98   
4             Pallas Athena, Venus and Juno     1593   
...                                     ...      ...   
51454                                 Angel     1682   
51455                         Exterior view  1854-62   
51456                         Exterior view  1854-62   
51457                         Exterior view  1854-62   
51458                         Exterior view  1839-43   

                         TECHNIQUE  \
0        Oil on canvas, 68 x 95 cm   
1        Oil on copper, 56 x 47 cm   
2      Oil on canvas, 197 x 142 cm   
3        Oil on copper, 31 x 21 cm   
4        Oil on canvas, 54 x 67 cm   
...                            ...   
51454                       Marble   
51455                        Photo   
51456                        Photo   
51457                        Photo   
51458                        Photo   

                                             LOCATION  \
0      Fogg Art Museum, Harvard University, Cambridge   
1                             Alte Pinakothek, Munich   
2                       The Hermitage, St. Petersburg   
3                    Kunsthistorisches Museum, Vienna   
4                         Museum of Fine Arts, Boston   
...                                               ...   
51454                      Abbey Church, Kremsmünster   
51455                     Castle Moyland, Bedburg-Hau   
51456                     Castle Moyland, Bedburg-Hau   
51457                     Castle Moyland, Bedburg-Hau   
51458                        St. Apollinaris, Remagen   

                                                     URL          FORM  \
0           https://www.wga.hu/html/a/aachen/adonis.html      painting   
1         https://www.wga.hu/html/a/aachen/allegory.html      painting   
2         https://www.wga.hu/html/a/aachen/allegorz.html      painting   
3          https://www.wga.hu/html/a/aachen/antiope.html      painting   
4           https://www.wga.hu/html/a/aachen/athena.html      painting   
...                                                  ...           ...   
51454  https://www.wga.hu/html/z/zurn/michael/angel.html     sculpture   
51455    https://www.wga.hu/html/z/zwirner/moyland1.html  architecture   
51456    https://www.wga.hu/html/z/zwirner/moyland2.html  architecture   
51457    https://www.wga.hu/html/z/zwirner/moyland3.html  architecture   
51458     https://www.wga.hu/html/z/zwirner/remagen.html  architecture   

               TYPE  SCHOOL  TIMEFRAME  
0      mythological  German  1601-1650  
1      mythological  German  1601-1650  
2      mythological  German  1601-1650  
3      mytholog

In [5]:
print(f"dataset length: {len(table)}")

dataset length: 51459


**As you can see the dataset has 51459 rows and 11 columns representing the metadata of each art.**  

### Analysis by SCHOOL 

For the purpose of web scraping, the dataset appears to be very big therefore a subset will be used. This subset will be chosen based on the SCHOOL and FORM 

In [9]:
listschool = table["SCHOOL"].unique()
dic = {}
for i in listschool:
    print (f"Number of {i} forms, {len(table[table.SCHOOL == i])}")
   

Number of German forms, 3949
Number of Danish forms, 215
Number of Spanish forms, 1853
Number of French forms, 7915
Number of Italian forms, 22588
Number of Dutch forms, 5205
Number of Scottish forms, 115
Number of Swedish forms, 74
Number of Flemish forms, 4722
Number of Netherlandish forms, 778
Number of Swiss forms, 233
Number of Russian forms, 217
Number of Greek forms, 91
Number of Other forms, 259
Number of American forms, 229
Number of English forms, 1448
Number of Austrian forms, 510
Number of Portuguese forms, 106
Number of Bohemian forms, 95
Number of Catalan forms, 111
Number of Hungarian forms, 450
Number of Irish forms, 108
Number of Norwegian forms, 32
Number of Polish forms, 46
Number of Belgian forms, 105
Number of Finnish forms, 5


**As we can see the highest number of FORMS is from the Italian SCHOOL and the lowest number is from the Finnish SCHOOL.** 

In [16]:
listschool = table["SCHOOL"].unique()  
for i in listschool:
    print (f"Number of {i} paintings, {len(table[(table.SCHOOL == i) & (table.FORM == 'painting')])}")
   

Number of German paintings, 1883
Number of Danish paintings, 125
Number of Spanish paintings, 1245
Number of French paintings, 4326
Number of Italian paintings, 13741
Number of Dutch paintings, 4374
Number of Scottish paintings, 74
Number of Swedish paintings, 37
Number of Flemish paintings, 3693
Number of Netherlandish paintings, 514
Number of Swiss paintings, 127
Number of Russian paintings, 111
Number of Greek paintings, 47
Number of Other paintings, 41
Number of American paintings, 178
Number of English paintings, 650
Number of Austrian paintings, 323
Number of Portuguese paintings, 25
Number of Bohemian paintings, 39
Number of Catalan paintings, 51
Number of Hungarian paintings, 345
Number of Irish paintings, 33
Number of Norwegian paintings, 24
Number of Polish paintings, 18
Number of Belgian paintings, 83
Number of Finnish paintings, 5


**Now it is filtered by the FORM "painting" and SCHOOL**

## French Paintings Dataset

Based on the analysis, I am creating my own dataset that I will be calling "French Paintings Dataset" the reason I chose the French **SCHOOL** over other schools, is because the amount of data related to the French SCHOOL and the **FORM** ```paintings``` is enough to be annotated and to be used as a machine learning model input.  

In [18]:
french_paintings = table[(table.SCHOOL == "French") & (table.FORM == 'painting')].reset_index()
french_paintings

index              AUTHOR  \
0       493  ALLEGRAIN, Etienne   
1       494  ALLEGRAIN, Etienne   
2       495  ALLEGRAIN, Etienne   
3      1304    ANGRAND, Charles   
4      1305    ANGRAND, Charles   
...     ...                 ...   
4321  51283         ZIEM, Félix   
4322  51284         ZIEM, Félix   
4323  51285         ZIEM, Félix   
4324  51286         ZIEM, Félix   
4325  51287         ZIEM, Félix   

                                             BORN-DIED  \
0                     (b. 1644, Paris, d. 1736, Paris)   
1                     (b. 1644, Paris, d. 1736, Paris)   
2                     (b. 1644, Paris, d. 1736, Paris)   
3     (b. 1854, Criquetot-sur-Ouville, d. 1926, Rouen)   
4     (b. 1854, Criquetot-sur-Ouville, d. 1926, Rouen)   
...                                                ...   
4321                 (b. 1821, Beaune, d. 1911, Paris)   
4322                 (b. 1821, Beaune, d. 1911, Paris)   
4323                 (b. 1821, Beaune, d. 1911, Paris)   
4324                 (b. 1821, Beaune, d. 1911, Paris)   
4325                 (b. 1821, Beaune, d. 1911, Paris)   

                                    TITLE     DATE  \
0     Landscape with the Finding of Moses        -   
1                         River Landscape        -   
2                         River Landscape        -   
3                    Couple in the Street     1887   
4       The Western Railway Leaving Paris     1886   
...                                   ...      ...   
4321         Riva degli Schiavone, Venice  c. 1847   
4322          San Simeone Piccolo, Venice     1865   
4323            The Rialto Bridge, Venice        -   
4324                     A View of Venice        -   
4325     Venice: Cannon Shot in the Basin        -   

                              TECHNIQUE  \
0            Oil on canvas, 88 x 115 cm   
1           Oil on canvas : 118 x 92 cm   
2           Oil on canvas : 86 x 136 cm   
3     Oil on canvas on card, 39 x 33 cm   
4             Oil on canvas, 75 x 92 cm   
...                                 ...   
4321         Oil on canvas, 68 x 105 cm   
4322          Oil on canvas, 54 x 69 cm   
4323           Oil on panel, 72 x 93 cm   
4324           Oil on panel, 60 x 81 cm   
4325          Oil on canvas, 50 x 83 cm   

                                   LOCATION  \
0             The Hermitage, St. Petersburg   
1     Musée National du Château, Versailles   
2                    Musée du Louvre, Paris   
3                      Musée d'Orsay, Paris   
4                        Private collection   
...                                     ...   
4321                     Private collection   
4322                     Private collection   
4323                     Private collection   
4324                     Private collection   
4325                     Private collection   

                                                    URL      FORM       TYPE  \
0     https://www.wga.hu/html/a/allegrai/etienne/lan...  painting  landscape   
1     https://www.wga.hu/html/a/allegrai/etienne/riv...  painting  landscape   
2     https://www.wga.hu/html/a/allegrai/etienne/riv...  painting  landscape   
3         https://www.wga.hu/html/a/angrand/couple.html  painting      genre   
4        https://www.wga.hu/html/a/angrand/railway.html  painting  landscape   
...                                                 ...       ...        ...   
4321          https://www.wga.hu/html/z/ziem/ziem6.html  painting  landscape   
4322          https://www.wga.hu/html/z/ziem/ziem7.html  painting  landscape   
4323          https://www.wga.hu/html/z/ziem/ziem8.html  painting  landscape   
4324          https://www.wga.hu/html/z/ziem/ziem9.html  painting  landscape   
4325         https://www.wga.hu/html/z/ziem/ziem91.html  painting  landscape   

      SCHOOL  TIMEFRAME  
0     French  1651-1700  
1     French  1651-1700  
2     French  1651-1700  
3     French  1851-1900  
4     French  1851-1900  
...      ...        ...  
4321  French  

In [19]:
french_paintings.to_csv("french_paintings_metadata.csv")